### PR TITLE
feat: Extract default devshell to devenv.nix

### PR DESCRIPTION
**Pull request by sestrella**

### CODE REVIEW

**Comment by oscar-izval** [comment]
Looks great @sestrella! Very happy to see devenv in the project. Just a few thoughts:
- Do you have some insights about using devenv on its own vs integrating it with Flakes? I guess there's an argument to be made about separation of concerns, but the integration is great and I see some benefits, specially not requiring two separate lock files with different versions of nixpkgs.
- We may want to add some notes about the use of devenv and direnv in the [Contributing](https://github.com/stackbuilders/nixpkgs-terraform/blob/main/README.md#contributing) section.

**Comment by sestrella** [comment]
@oscar-izval thank you for the provided feedback. Regarding:

> Do you have some insights about using devenv on its own vs integrating it with Flakes? I guess there's an argument to be made about separation of concerns, but the integration is great and I see some benefits, specially not requiring two separate lock files with different versions of nixpkgs.

I guess the main motivation here is that the existing devshell is primarily used to set up all of the required tools for the update script; in that sense, it is not something that affects the end users of this flake, but rather something we use as maintainers that, in my opinion, should not be exposed in the flake.nix file; I believe this was my main motivation for extracting the devshell from the original file and configuring devenv. Whether we use devenv on its own or as part of a flake integration, I believe it depends on whether there is a valid use case for providing a devshell to the consumers of this flake in the first place. In this case, I would argue that keeping it as a standalone Nix file makes more sense because the provided shell is only used for maintenance.

> We may want to add some notes about the use of devenv and direnv in the [Contributing](https://github.com/stackbuilders/nixpkgs-terraform/blob/main/README.md#contributing) section.

That was a good call! If you agree with this change, I would be happy to update the documentation.

File: .github/workflows/build.yml
Comment (oscar-izval) [inline .github/workflows/build.yml:29]: It's not immediately obvious to be what this job is doing by the name, but I'm not super familiar with `devenv ci` so I'll defer to you on this one. I guess it would be clear if there was a formatting error, but wanted to bring it up in case you can think of a better name.

**Comment by oscar-izval** [comment]
After reading your comment looks like the best approach is to keep them separate, thanks for the context!

**Comment by sestrella** [comment]
@oscar-izval would you mind taking a second look, please?

File: devenv.yaml
Comment (oscar-izval) [inline devenv.yaml:14]: Should we get rid if these defaults? I'm not sure they apply to the development workflow of this project

File: devenv.nix
Comment (oscar-izval) [inline devenv.nix:23]: Similar thing here, wdyt about removing this comment?

**Review by oscar-izval — APPROVED** [review]
LGTM! Maybe just remove those comments, but it's not a big deal. Something to consider is that we were exposing that devshell, and even if nobody uses it, it should technically be a major release. Should we do that alongside the removal of the versions-by-cycle code?

**Comment by sestrella** [comment]
@oscar-izval I merged this PR but canceled the build to avoid generating a major release right now, as I am planning to open a new PR that removes the cycle packages.

**Pull request merged**
